src/test/java/htsjdk/samtools/BAMFileWriterTest.java

```diff
@@ -60,50 +59,68 @@ private SAMRecordSetBuilder getRecordSetBuilder(final boolean sortForMe, final S
         }
+
+        File tempMetrics = File.createTempFile("CGTagTest", ".validation_metrics");
+        try (final SamReader samReader = SamReaderFactory.makeDefault().open(bamFile)) {
+
+            new SamFileValidator(new PrintWriter(new FileOutputStream(tempMetrics)),
+                    100).validateSamFileSummary(samReader, null);
+        }
+
+        final MetricsFile<SamFileValidator.ValidationMetrics, String> validationMetrics = new MetricsFile<>();
+        validationMetrics.read(new FileReader(tempMetrics));
+
+        Assert.assertTrue(validationMetrics.getHistogram().get("ERROR:CG_TAG_FOUND_IN_ATTRIBUTES") == null);
+    }
+
+    private void verifyBAMFile(final SAMRecordSetBuilder samRecordSetBuilder, final File bamFile) throws IOException {
+        try (final SamReader bamReader = SamReaderFactory.makeDefault().open(bamFile);
+             final SamReader samReader = samRecordSetBuilder.getSamReader()) {
+            verifySamReadersEqual(samReader, bamReader);
+        }
     }
 
-    private void verifyBAMFile(final SAMRecordSetBuilder samRecordSetBuilder, final File bamFile) {
+    private void verifySamReadersEqual(final SamReader reader2, final SamReader reader1) {
 
-        final SamReader bamReader = SamReaderFactory.makeDefault().open(bamFile);
-        final SamReader samReader = samRecordSetBuilder.getSamReader();
-        samReader.getFileHeader().setSortOrder(bamReader.getFileHeader().getSortOrder());
-        Assert.assertEquals(bamReader.getFileHeader(), samReader.getFileHeader());
-        final CloseableIterator<SAMRecord> it = samReader.iterator();
-        final CloseableIterator<SAMRecord> bamIt = bamReader.iterator();
-        while (it.hasNext()) {
-            Assert.assertTrue(bamIt.hasNext());
-            final SAMRecord samRecord = it.next();
-            final SAMRecord bamRecord = bamIt.next();
+        reader2.getFileHeader().setSortOrder(reader1.getFileHeader().getSortOrder());
+        Assert.assertEquals(reader1.getFileHeader(), reader2.getFileHeader());
 
-            // SAMRecords don't have this set, so stuff it in there
-            samRecord.setIndexingBin(bamRecord.getIndexingBin());
+        try (final CloseableIterator<SAMRecord> samIt1 = reader1.iterator();
+             final CloseableIterator<SAMRecord> samIt2 = reader2.iterator()) {
+            while (samIt2.hasNext()) {
+                Assert.assertTrue(samIt1.hasNext());
+                final SAMRecord samRecord1 = samIt1.next();
+                final SAMRecord samRecord2 = samIt2.next();
 
-            // Force reference index attributes to be populated
-            samRecord.getReferenceIndex();
-            bamRecord.getReferenceIndex();
-            samRecord.getMateReferenceIndex();
-            bamRecord.getMateReferenceIndex();
+                // SAMRecords don't have this set, so stuff it in there
+                samRecord2.setIndexingBin(samRecord1.getIndexingBin());
```

Comment:
I think this line is masking a bug that's resetting the indexing bin.  I'm not sure what to do about that in testing though.  Maybe we should have a test specifically for the indexing bin.